File: auth-service/src/main/java/com/pouetteur/authservice/dto/ProfileDTO.java
package com.pouetteur.authservice.dto;

import com.pouetteur.authservice.model.Reaction;
import com.pouetteur.authservice.model.Role;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProfileDTO implements Cloneable{

    private String id;
    private String username;
    private String name;
    private Date creationDate;
    private String profilePhoto;
    private String profileBanner;
    private String bio;
    private List<ProfileDTO> followings;
    private List<ProfileDTO> followers;
    private Map<Reaction, Integer> reactions;
    private String email;
    private List<Role> roles;
    private String birthdate;
    private List<ProfileDTO> members;
    private ProfileDTO owner;
    private String classType;

    @Override
    public ProfileDTO clone() {
        try {
            return (ProfileDTO) super.clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: profile-service/profile-service/DTOs/ProfileDTO.cs
using System.ComponentModel.DataAnnotations;
using profile_service.Models;

namespace profile_service.DTOs;

public class ProfileDto
{
    [Key]
    public string? Id { get; set; }
    [Required]
    public string? Username { get; set; }
    public string? Name { get; set; }
    public DateTime? CreationDate { get; set; }
    public string? ProfilePhoto { get; set; }
    public string? ProfileBanner { get; set; }
    public string? Bio { get; set; }
    public List<ProfileDto>? Followings { get; set; }
    public List<ProfileDto>? Followers { get; set; }
    public Dictionary<Reaction,int>? Reactions { get; set; }
    public string? Email { get; set; }
    public string? Roles { get; set; }
    public DateTime? Birthdate { get; set; }
    public List<ProfileDto>? Members { get; set; }
    public ProfileDto? Owner { get; set; }
}